core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -337,4 +394,67 @@ protected String tableName() {
       return tableIdentifier.toString();
     }
   }
+
+  private class InMemoryViewOperations extends BaseViewOperations {
+    private final FileIO fileIO;
+    private final TableIdentifier identifier;
+
+    InMemoryViewOperations(FileIO fileIO, TableIdentifier identifier) {
+      this.fileIO = fileIO;
+      this.identifier = identifier;
+    }
+
+    @Override
+    public void doRefresh() {
+      String latestLocation = views.get(identifier);
+      if (latestLocation == null) {
+        disableRefresh();
+      } else {
+        refreshFromMetadataLocation(latestLocation);
+      }
+    }
+
+    @Override
+    public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+      String newLocation = writeNewMetadata(metadata, currentVersion() + 1);
+      String oldLocation = base == null ? null : currentMetadataLocation();
```

Comment:
This is incorrect because there is no guarantee that `base` and `currentMetadataLocation` match. `ViewMetadata` needs to carry the location it was loaded from so it can be used here.

Reply:
I added `currentMetadataLocation` tracking to `ViewMetadata`, so this should be fixed now